### DISCUSSION TITLE
Support for custom query parameters in WMS/WFS GetCapabilities or token authentication URL parameters

### DISCUSSION THREAD

**Discussion by Kate-Lyndegaard** [INSPIRE-MIF/helpdesk-validator#1015] [Ideas]
We have WMS and WFS services that require token authentication provided in the URL.

Here is an example URL with token authentication: 
[https://haleconnect.com/ows/services/org.1326.b6b2f6ed-14da-4657-9e8e-e15538fd6d57_w[…]PeJ5vXhsyZpg34CjFY3ZBWgZwUgVG1SmoUFzMLeGCICihCrYVWY-xpa4qh7bQ](https://haleconnect.com/ows/services/org.1326.b6b2f6ed-14da-4657-9e8e-e15538fd6d57_wms?SERVICE=WMS&REQUEST=GetCapabilities&VERSION=1.3.0&kelvin_token=eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiIxMzI2IiwiZXhwIjoxNzAyOTczODUzfQ.Xw26pl4L7WJ5srMGyn6Gf1gOIPeJ5vXhsyZpg34CjFY3ZBWgZwUgVG1SmoUFzMLeGCICihCrYVWY-xpa4qh7bQ)

Currently, these services fail the WMS and WFS conformance classes because the token is not included in the validator requests. We would like to request that the validator support services which require token authentication, or alternatively custom query parameters in the WMS/WFS GetCapabilities.

**Comment by jenriquesoriano** [comment]
Dear @Kate-Lyndegaard ,

thank you for your enhancement suggestion.
In order to collect feedback from the community, we have converted this issue to a discussion, so we can obtain further input from the community.

Best regards,

**Comment by lvgl-cs** [comment]
Hello,

I wanted to express my support for this request.
I am currently facing a similar challenge in my role at the GDI Saarland while working with our Geoportal. Our WMS/WFS services also require additional parameters. For WFS we have two custom URL parameters, namely INSPIRE and FEATURETYPE_ID. An example URL from our Geoportal looks like this: https://geoportal.saarland.de/mapbender/php/wfs.php?INSPIRE=1&FEATURETYPE_ID=1536&REQUEST=GetCapabilities&SERVICE=WFS&VERSION=2.0.0

Our services also fail because the parameters are not considered in the validator requests. 
An enhancement to include these parameters would be of great benefit for us which is why we would appreciate any developments in this regard.

Best regards,
Carsten Sommer (GDI Saarland)
